client/src/main/java/com/google/cloud/pubsub/AbstractSubscriberConnection.java

```diff
@@ -310,7 +310,10 @@ public void run() {
 
   private void addOutstadingAckHandlers(
       ExpirationInfo expiration, final List<AckHandler> ackHandlers) {
-    outstandingAckHandlers.putIfAbsent(expiration, new ArrayList<AckHandler>(ackHandlers.size()));
```

Comment:
good question, based on [java doc](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#putIfAbsent-K-V-), default imlementation doesn't have such guarantees. 

@davidtorresv do you have any idea or preferences on this change?